src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -70,19 +96,18 @@ public ColumnAttribute(string ordinal, string name = null)
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
     public sealed class ColumnNameAttribute : Attribute
     {
-        private readonly string _name;
         /// <summary>
         /// Column name.
         /// </summary>
-        public string Name { get { return _name; } }
+        internal string Name { get; }
 
         /// <summary>
         /// Allows one to specify a name to expose this column as, as opposed to simply
         /// the field name.
```

Comment:
The attribute may be applied to a property as well:
```suggestion
        /// the field or property name.
```